### ISSUE TITLE
Deobfuscation goes crazy

### ISSUE THREAD

**Issue by jpstotz** [skylot/jadx#1364]
In the last days I observed multiple times that the deobfuscation goes crazy:

1. Class names in the AndroidManifest.xml were modified but the class names in the source tree were original. Example:
  * Class name in AndroidManifest.xml: `com.sec.android.app.myfiles.external.p011ui.MainActivity`
  * Class name in source tree: `com.sec.android.app.myfiles.external.ui.MainActivity`

The modified class name in AndroidManifest.xml even remained after turning of deobfuscation. Happened with [this APK](https://apkpure.com/de/samsung-my-files/com.sec.android.app.myfiles/download/1011001342-APK?from=versions%2Fversion) but only if deobfuscation is active when Jadx starts.

2. Package names are deobfuscated where it is not necessary:
Package `com.google.android.exoplayer2` becomes `google.google.android.exoplayer2`. 

3. Other strange renamings:
`com.worldturner.medeia.parser.gson.GsonJsonReaderDecoratorKt` becomes `google.worldturner.medeia.java.math.BigInteger`
In the file I found the comment `reason: invalid package`.

Point 2 and 3 I observed in [this APK](https://apkpure.com/de/crunchyroll/com.crunchyroll.crunchyroid/download/540-APK-7a8a309939a14b17ddd8dfebe87ef2e6?from=versions%2Fversion)

**Comment by jpstotz** [comment]
@skylot Github was a little bit too fast in saving this issue. I just had started to write it when it somehow got saved...

**Comment by skylot** [comment]
Looks like most issues can be caused by https://github.com/skylot/jadx/blob/9ef99a2b92a76fc9d26a0e9501ccb7c1ec608add/jadx-core/src/main/java/jadx/core/xmlgen/XmlDeobf.java#L17
this map is static and never cleared or updated after init :worried:

@jpstotz  do these crazy things happen only in AndroidManifest.xml or also appear in code?

**Comment by jpstotz** [comment]
> do these crazy things happen only in AndroidManifest.xml or also appear in code?

It seems this is not deterministic. At the first time I observed the first problem it was AndroidManifest.xml only I think (I noticed this because I wasn't able to go to the actual class by clicking it in the manifest). 
When I now open that APK the class/package has also been renamed.

Problem 2 and 3 I have seen both in code and the source tree of Jadx. For problem 2 & 3 it seems like the deobfuscation places maps the wrong parts of the full qualified class name (my guess form the Kotlin deobfuscation class name source) to non-matching parts of the original class name (like mapping `google` to `the `com` part of `com.google`).
